gpu-alloc/src/linear.rs

```diff
@@ -231,12 +226,21 @@ where
 
     #[cfg_attr(feature = "tracing", tracing::instrument(skip(self, device)))]
     pub unsafe fn cleanup(&mut self, device: &impl MemoryDevice<M>) {
-        if let Some(chunk) = self.ready.take() {
-            assert_eq!(
-                chunk.allocated, 0,
+        if let Some(mut chunk) = self.ready.take() {
+            debug_assert!(
+                is_unique(&mut chunk.memory),
                 "All blocks must be deallocated before cleanup"
             );
-            device.deallocate_memory(chunk.memory);
+
+            match Arc::try_unwrap(chunk.memory) {
```

Comment:
it's redundant to call `is_unique` only to follow by `try_unwrap`. They do the same thing

Reply:
The intent was to panic in debug here and merely whining into traces in release

Reply:
you can still do this without `is_unique`, can't you?
```rust
Err(_) => {
  debug_assert!(false, "All blocks must be deallocated before cleanup"),
  #[cfg(feature = "tracing")]
  tracing::error!("Cleanup before all memory blocks are deallocated");
}
``